pkg/kademlia/kademlia.go

```diff
@@ -451,8 +452,19 @@ func (k *Kad) AddPeers(ctx context.Context, addrs ...swarm.Address) error {
 	return nil
 }
 
+func (k *Kad) Pick(peer p2p.Peer) bool {
+	po := swarm.Proximity(k.base.Bytes(), peer.Address.Bytes())
+	_, oversaturated := k.saturationFunc(po, k.knownPeers, k.connectedPeers)
+	return oversaturated
```

Comment:
Kademlia pick function would return true here if the bin is oversaturated. Is this expected behaviour for pick? Especially in the context of usage in libp2p notifier in handshake handler.

Reply:
nice catch. added this part to the unit test that checks the relevant functionality